docs/types.rst

```diff
@@ -1108,10 +1114,62 @@ Of course, you can also directly access the members of the struct without
 assigning it to a local variable, as in
 ``campaigns[campaignID].amount = 0``.
 
+Types as input and output parameters
+====================================
+
+.. index:: !function parameters, functions, parameters
+
+Allowed function parameter types
+--------------------------------
+
+**All** :ref:`value type <value-types>` variables are allowed as input and
+output function parameter types.
+
+**All** :ref:`reference type <reference-types>` variables are allowed as input
+and output function parameter types for internal functions, and for external
+functions if you enable the experimental ``ABIEncoderV2`` feature.
+
+:ref:`Mapping type <mapping-types>` variables **are not** allowed as input or output function types.
```

Comment:
except for internal ones or for libraries

Reply:
This is why I wanted to make a table :)

I'll try and think of a way to make these clear to understand.

Reply:
Hang on @chriseth I am slightly confused now, as far as I can tell, this comment and comment https://github.com/ethereum/solidity/pull/4859/commits/69f446966efd3f15246ad395b4753133fa050ef6#r222998716 contradict each other. 

Or you just mean in this comment, that mapping types have the additional caveat of those that come from libraries, or that enabling `ABIEncoderV2` still doesn't make them available? Or that mapping types just have their own rules that you mention in your first comment here?